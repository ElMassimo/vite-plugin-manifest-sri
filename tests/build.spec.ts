import { readFileSync } from 'fs'
import { resolve } from 'path'
import spawn from 'cross-spawn'
import { beforeAll, describe, test, expect } from 'vitest'

const exampleRoot = resolve(__dirname, '../example')

describe('build', () => {
  beforeAll(() => buildApp(), 5000)

  test('adds integrity hash', () => {
    const manifest = readManifest('manifest.json')
    expect(manifest['entrypoints/main.ts'].integrity)
      .toEqual('sha384-vJj4PmZ9vllU99cNXZ5ozrfo6a0h/wUVLWsVtmbltVr32o549nSSXYMj8ZzenCLt')

    expect(manifest['dynamic.ts'].integrity)
      .toEqual('sha384-KorBstcdnwE8bqEM96n5weR+Q9mNUAR0Hr7tewyfRadn5v1IY58p2bfjmnkxTotE')

    expect(manifest['entrypoints/styles.css'].integrity)
      .toEqual('sha384-HmlJ0WJXVNLaqWQYVRhZgsW1JBN4XBjm3j+j38nagNFimvBxi8nZ1FUysAc9JJvL')

    expect(manifest['entrypoints/main.ts'].cssIntegrity[0])
      .toEqual('sha384-bqzLq+mQSriGZYcwm8aYVPI+TilL6LHvyx/zGwJ8t3r08UKKjeIkeleDKAYTZT1J')
  })

  test('manifest snapshot', () => {
    expect(readAsset('manifest.json')).toMatchSnapshot()
    expect(readAsset('manifest-assets.json')).toMatchSnapshot()
  })
})

function readManifest (path: string) {
  return JSON.parse(readAsset(path))
}

function readAsset (path: string) {
  return readFileSync(`${exampleRoot}/public/vite/.vite/${path}`, 'utf-8')
}

function buildApp () {
  spawn.sync('npm', ['run', 'build'], { stdio: process.env.DEBUG ? 'inherit' : undefined, cwd: exampleRoot })
}
