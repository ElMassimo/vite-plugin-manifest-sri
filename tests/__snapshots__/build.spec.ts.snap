// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`build > manifest snapshot 1`] = `
"{
  \\"dynamic.ts\\": {
    \\"file\\": \\"assets/dynamic-y-YcnQ4I.js\\",
    \\"isDynamicEntry\\": true,
    \\"src\\": \\"dynamic.ts\\",
    \\"integrity\\": \\"sha384-KorBstcdnwE8bqEM96n5weR+Q9mNUAR0Hr7tewyfRadn5v1IY58p2bfjmnkxTotE\\"
  },
  \\"entrypoints/main.ts\\": {
    \\"css\\": [
      \\"assets/main-1h5ju1v-.css\\"
    ],
    \\"dynamicImports\\": [
      \\"dynamic.ts\\"
    ],
    \\"file\\": \\"assets/main-dLnuJ4t2.js\\",
    \\"isEntry\\": true,
    \\"src\\": \\"entrypoints/main.ts\\",
    \\"integrity\\": \\"sha384-vJj4PmZ9vllU99cNXZ5ozrfo6a0h/wUVLWsVtmbltVr32o549nSSXYMj8ZzenCLt\\"
  },
  \\"entrypoints/styles.css\\": {
    \\"file\\": \\"assets/styles-5G55TvLQ.css\\",
    \\"isEntry\\": true,
    \\"src\\": \\"entrypoints/styles.css\\",
    \\"integrity\\": \\"sha384-HmlJ0WJXVNLaqWQYVRhZgsW1JBN4XBjm3j+j38nagNFimvBxi8nZ1FUysAc9JJvL\\"
  }
}"
`;

exports[`build > manifest snapshot 2`] = `"{}"`;
